### PR TITLE
Add management tests

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

File: management/src/main/java/org/kaazing/gateway/management/filter/FullManagementFilterStrategy.java [hunk 1]
```diff
@@ -61,7 +61,6 @@ public void doSessionClosed(ManagementContext managementContext,
         if (sessionBean != null) {
             managementContext.removeSessionManagementBean(sessionBean);
             managementContext.getManagementSessionStrategy().doSessionClosed(sessionBean);
-            managementContext.getManagementSessionStrategy().doSessionClosed(sessionBean);
         }
```
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/filter/FullManagementFilterStrategy.java]: No need to do this twice.  I think it was added due to notifications not being delivered properly.  This doesn't fix that and has the resulting NPE appear twice.

File: management/src/main/java/org/kaazing/gateway/management/jmx/JmxManagementServiceHandler.java [hunk 2]
```diff
@@ -178,13 +178,14 @@ public void addSessionManagementBean(SessionManagementBean sessionManagementBean
                             JMX_OBJECT_NAME,
                             gatewayManagementBean.getHostAndPid(),
                             replaceCharactersDisallowedInObjectName(serviceManagementBean.getServiceType()),
-                            address.getExternalURI(),
+                            serviceManagementBean.getServiceName(),
```
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/jmx/JmxManagementServiceHandler.java]: We now register on service name

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java [hunk 3]
```diff
@@ -119,22 +124,25 @@ public void getRoundTripLatencyAttributesFromJmx() throws Exception {
...
 
-        k3po.notifyBarrier("READ_LATENCY_ATTRIBUTES");
-
         assertTrue("Could not retrieve Round Trip Latency from Jmx", latency > -1);
         assertTrue("Could not retrieve Round Trip Latency Timestamp from Jmx", latencyTimestamp > currentTimestamp);
-
         k3po.finish();
     }
 
+    @Ignore("This test should be written in HTTP so its generic and doesn't break as layers change,"
```
Comment (dpwspoon) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java]: @sbadugu Can you file a ticket to track this

File: management/src/main/java/org/kaazing/gateway/management/filter/FullManagementFilterStrategy.java (hunk 1 above)
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/filter/FullManagementFilterStrategy.java]: Also it is not there in 4.x

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java (hunk 3 above)
Comment (sbadugu) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java]: It's being tracked [here](https://github.com/kaazing/tickets/issues/413).

**Comment by dpwspoon** [comment]
This is passing for me though I did see it hang once with what appeared to be a K3po bug on dispose.  I can't reproduce.

>
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:28 min
[INFO] Finished at: 2016-04-05T11:07:19-07:00

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java [hunk 4]
```diff
@@ -119,22 +124,23 @@ public void getRoundTripLatencyAttributesFromJmx() throws Exception {
 
         MBeanServerConnection mbeanServerConn = jmxConnection.getConnection();
         Set<ObjectName> mbeanNames = mbeanServerConn.queryNames(null, null);
-        String MBeanPrefix = "subtype=services,serviceType=echo,serviceId=\"" + WS_URI + "\",name=sessions";
+        String MBeanPrefix = "subtype=services,serviceType=echo,serviceId=\"" + ECHO_WSN_SERVICE + "\",name=sessions";
         for (ObjectName name : mbeanNames) {
             if (name.toString().indexOf(MBeanPrefix) > 0) {
+                mbeanServerConn.getObjectInstance(name);
```
Comment (jitsni) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java]: Is this needed ?

**Comment by jitsni** [comment]
Looks ok.

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java (hunk 4 above)
Comment (dpwspoon) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRoundTripLatencyIT.java]: I think so, though I'm no JMX expert.  In the debugger this invoked the RMI to get the instance locally, the following commands queried it

**Comment by dpwspoon** [comment]
Thanks @jitsni, I'm waiting for Travis CI to go through as it needs the truststore set up quickly.  IF it does I'll merge it

**Pull request merged**